pkg/solana/txm/pendingtx.go

```diff
@@ -81,20 +99,20 @@ type pendingTxContext struct {
 
 func newPendingTxContext() *pendingTxContext {
 	return &pendingTxContext{
-		cancelBy: map[string]context.CancelFunc{},
-		sigToID:  map[solana.Signature]string{},
+		cancelBy:    map[string]context.CancelFunc{},
+		sigToTxInfo: map[solana.Signature]txInfo{},
 
 		broadcastedProcessedTxs: map[string]pendingTx{},
 		confirmedTxs:            map[string]pendingTx{},
 		finalizedErroredTxs:     map[string]finishedTx{},
 	}
 }
 
-func (c *pendingTxContext) New(tx pendingTx, sig solana.Signature, cancel context.CancelFunc) error {
+func (c *pendingTxContext) New(tx pendingTx) error {
 	err := c.withReadLock(func() error {
-		// validate signature does not exist
-		if _, exists := c.sigToID[sig]; exists {
-			return ErrSigAlreadyExists
+		// Check if ID already exists in any of the maps
+		if _, exists := c.broadcastedProcessedTxs[tx.id]; exists {
```

Comment:
Now that you're using tx.id for already existing entries, won't that cause unnecessary errors when calling this function from rebroadcastExpiredTxs()?

I see that rebroadcastExpiredTxs() will collect all expired txs, and then call txm.sendWithRetry() on them, using the same tx.id. That looks to me will fail because this function will return the error ErrIDAlreadyExists.

Reply:
It should be handled gracefully. The `rebroadcastExpiredTxs` function calls [rebroadcastWithGivenBlockhash method in this line](https://github.com/smartcontractkit/chainlink-solana/blob/nonevm-984-reorg/pkg/solana/txm/txm.go#L629) where we [remove the previous tx from state](https://github.com/smartcontractkit/chainlink-solana/blob/404708355148ebdfdd9737b1d46d8d6393c72e89/pkg/solana/txm/txm.go#L963) before calling [sendWithRetry](https://github.com/smartcontractkit/chainlink-solana/blob/404708355148ebdfdd9737b1d46d8d6393c72e89/pkg/solana/txm/txm.go#L975)